beacon-chain/cache/registration.go

```diff
@@ -0,0 +1,82 @@
+package cache
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v4/config/params"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v4/encoding/bytesutil"
+	"github.com/prysmaticlabs/prysm/v4/math"
+	ethpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
+	log "github.com/sirupsen/logrus"
+	"go.opencensus.io/trace"
+)
+
+// RegistrationCache is used to store the cached results of an Validator Registration request.
+// beacon api /eth/v1/validator/register_validator
+type RegistrationCache struct {
+	indexToRegistration map[primitives.ValidatorIndex]*ethpb.ValidatorRegistrationV1
+	lock                sync.RWMutex
+}
+
+// NewRegistrationCache initializes the map and underlying cache.
+func NewRegistrationCache() *RegistrationCache {
+	return &RegistrationCache{
+		indexToRegistration: make(map[primitives.ValidatorIndex]*ethpb.ValidatorRegistrationV1),
+		lock:                sync.RWMutex{},
```

Comment:
deepsource complained the lock was open on the struct so now i added it internally here, can revert to align with other caches.